### ISSUE TITLE
Automation apply error

### ISSUE THREAD

**Issue by jbnitorum** [PrefectHQ/terraform-provider-prefect#338]
I am getting the following error when applying an imported automation.  There seems to be some issue with the 'expect' field in the trigger event.  It seems there is some inconsistency with the ordering that is causing errors.  I had successfully applied one of the automations but when i went to replan to try to change the order of the other one I am now seeing a similar issue on the one that worked.

So I switched those back and was able to apply that change successfully again but I assume if I replan I'll get that error.  It seems like the API and terraform are reordering the list of items and doing it in some different way.

I've tried reordering the states in various order but I still get errors, though the number of errors varies, so there might be some magical order out there that works.

Plan:

```diff
  # prefect_automation.xxxxxx will be updated in-place
! resource "prefect_automation" "xxxxxxx" {
        id                 = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
        name               = "xxxxxxxxxxxxxxx"
!       trigger            = {
!           event = {
!               after         = [
+                   "prefect.flow-run.Cancelled",
+                   "prefect.flow-run.Completed",
                    "prefect.flow-run.Crashed",
                    "prefect.flow-run.Failed",
-                   "prefect.flow-run.Completed",
-                   "prefect.flow-run.Cancelled",
                ]
                # (7 unchanged attributes hidden)
            }
        }
!       updated            = "2024-12-18T18:44:40Z" -> (known after apply)
        # (5 unchanged attributes hidden)
    }
```

apply error:

```
Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to prefect_automation.xxxxxxxxxxxxx,
│ provider "provider[\"registry.opentofu.org/prefecthq/prefect\"]" produced
│ an unexpected new value: .trigger.event.after[0]: was
│ cty.StringVal("prefect.flow-run.Cancelled"), but now
│ cty.StringVal("prefect.flow-run.Crashed").
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to prefect_automation.xxxxxxxxxxxxxxxxx,
│ provider "provider[\"registry.opentofu.org/prefecthq/prefect\"]" produced
│ an unexpected new value: .trigger.event.after[1]: was
│ cty.StringVal("prefect.flow-run.Completed"), but now
│ cty.StringVal("prefect.flow-run.Failed").
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to prefect_automation.xxxxxxxxxxxxxxxxxxxxx,
│ provider "provider[\"registry.opentofu.org/prefecthq/prefect\"]" produced
│ an unexpected new value: .trigger.event.after[2]: was
│ cty.StringVal("prefect.flow-run.Crashed"), but now
│ cty.StringVal("prefect.flow-run.Completed").
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to prefect_automation.xxxxxxxxxxxxxxxxx,
│ provider "provider[\"registry.opentofu.org/prefecthq/prefect\"]" produced
│ an unexpected new value: .trigger.event.after[3]: was
│ cty.StringVal("prefect.flow-run.Failed"), but now
│ cty.StringVal("prefect.flow-run.Cancelled").
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```

**Comment by jbnitorum** [comment]
Sorry, I said the 'expect' field but its actually the 'after' field in this example.  I saw a similar error also in the expect field.  Just using the CLI to inspect the automation looks like the API returns these lists in a random order.  Each time you run:

```
prefect automation inspect xxxxxx
````

the list fields come back in a different order

**Comment by parkedwards** [comment]
@jbnitorum we've just cut a new patch release with this fix, could you give this a whirl with 2.13.3 when it's up in the TF registry?

https://github.com/PrefectHQ/terraform-provider-prefect/releases/tag/v2.13.3
